add docs for JS debugger in language docs (#165)
spinframework/spin-docs: content/v3/javascript-components.md

{{ details "Unique ID generator" "- [nanoid](https://www.npmjs.com/package/nanoid)\n- [ulidx](https://www.npmjs.com/package/ulidx)\n- [uuid](https://www.npmjs.com/package/uuid)" }} 

> ## Debugging in VSCode

Author: @tschneidereit do you have any thoughts here?

The [StarlingMonkey Debugger](https://marketplace.visualstudio.com/items?itemName=BytecodeAlliance.starlingmonkey-debugger) enables debugging components built using the SDK. The debugger currently only works with the HTTP trigger and needs to be restarted for every request.

### Using the Debugger

#### Setup

> First, make sure that the extension is installed. Next verify that the templates are new enough to support debugging. This can be verified by looking `.vscode/setting.json` and verifying if something like the following is configured:

Reviewer: We can document the current versions of the templates and VS Code as the setup that works here and that we support.

> First, make sure that the extension is installed. Next verify that the templates are new enough to support debugging. This can be verified by looking `.vscode/setting.json` and verifying if something like the following is configured:

Reviewer 2: Maybe split this into "installing the extension" (once) and "setting up the project" (each project).

```json
// .vscode/setting.json
{
    "starlingmonkey": {
        "componentRuntime": {
            "executable": "spin",
            "options": [
                "up",
                "-f",
                "${workspaceFolder}",
            ],
        }
    }
}
```

If it does not exist, setup the configuration using the [testcase](https://github.com/spinframework/spin-js-sdk/tree/main/test/debugger-testing/spin-ts/.vscode) as a reference. 

Once that is setup, we need to update the configuration in `spin.toml` to build the debug component. This can be done by updating the build command in `spin.toml` to be `npm run build:debug`. We also need to add `"tcp://127.0.0.1:*"` to the list of `allowed_outbound_hosts`. 

Now we can build the app using the familiar `spin build` command.

#### Attaching the Debugger

> We can start the app and attach the debugger to it using the `F5` key or using the start debugger button as show below:

Reviewer 2: ```suggestion
We can start the app and attach the debugger to it using the `F5` key or using the Start Debugger button as show below:
```

![Starting VScode debugger](/static/image/docs/js-debugger.jpg)

> Once the debugger is attached, it can used as it normally would to do things like set breakpoints and step through the code. 

Reviewer 3: ```suggestion
Once the debugger is attached, it can be used as it normally would be. E.g., you can do things like setting breakpoints and stepping through the code.
```

- All `spin-sdk` related functions and methods (like `Variables`, `Redis`, `Mysql`, `Pg`, `Kv` and `Sqlite`) can be called only inside the fetch event handler. This includes `fetch`. Any attempts to use it outside the function will lead to an error. This is due to Wizer using only Wasmtime to execute the script at build time, which does not include any Spin SDK support.
- No crypto operation that involve handling private keys are supported. 

## Debugging in VSCode

You can use the experimental [StarlingMonkey Debugger](https://marketplace.visualstudio.com/items?itemName=BytecodeAlliance.starlingmonkey-debugger) to debug JavaScript HTTP components.

### Using the Debugger

> The debugger is a work in progress, and has some known issues and limitations.  In particular, you will need to restart it for each request.

#### Setup

> ##### Installing the Extension

Reviewer 2: Heading level 5?!?!?!  Let's reassess the structure here.  Maybe

```
## Debugging in VS Code
### Debugger Status
### Installing the Debugger
### Setting up a Project for Debugging
### Running a Project in the Debugger
```

(I don't feel that the "debugger is a work in progress" para needs to be a separate heading, this is just to retain the way you have it.  For myself, I would make that note part of the front matter.)

            "options": [
                "up",
                "-f",
                "${workspaceFolder}",
            ],
        }
    }
}

// .vscode/launch.json
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "starlingmonkey",
            "request": "launch",
            "name": "Debug StarlingMonkey component",
            "component": "${workspaceFolder}/dist/<component_name>.wasm",
            "program": "${workspaceFolder}/src/index.ts",
            "stopOnEntry": false,
            "trace": true
        }
    ]
}
```

If it does not exist, setup the configuration using the [testcase](https://github.com/spinframework/spin-js-sdk/tree/main/test/debugger-testing/spin-ts/.vscode) as a reference for setting up both the `settings.json` and `launch.json`. Make sure to setup the `program` field in `launch.json` based on whether it is a JavaScript or TypeScript project.

Once that is set up, we need to update the configuration in `spin.toml` to build the debug component. This can be done by setting the build command in `spin.toml` to be `npm run build:debug`. We also need to add `"tcp://127.0.0.1:*"` to the list of `allowed_outbound_hosts`. 

> Now we can build the app using the familiar `spin build` command.

Reviewer 2: ```suggestion
Now you can build the app using the familiar `spin build` command.
```

(ignore if you decide to standardise on 'we' rather than addressing the reader directly)

#### Attaching the Debugger

> We can start the app and attach the debugger to it using `F5` key or using the Start Debugger button as show below:

Reviewer 2: ```suggestion
You can start the app and attach the debugger to it using the `F5` key or using the Start Debugger button as show below:
```

![Starting VScode debugger](/static/image/docs/js-debugger.jpg)

> Once the debugger is attached, it can be used as it normally would be. E.g., you can do things like setting breakpoints and stepping through the code.

Reviewer 2: ```suggestion
Once the debugger is attached, it can be used in the same way as the built-in VS Code debugger. For example, you can do things like setting breakpoints and stepping through the code.
```
